hnn_core/network.py

```diff
@@ -1070,11 +1101,87 @@ def add_connection(self, src_gids, target_gids, loc, receptor,
             _validate_type(item, (int, float), arg_name, 'int or float')
             conn['nc_dict'][key] = item
 
+        # Probabilistically define connections
+        if probability != 1.0:
+            self._connection_probability(conn, probability, seed)
+
+        conn['probability'] = probability
+
         self.connectivity.append(deepcopy(conn))
 
+    def _connection_probability(self, conn, probability, seed=0):
```

Comment:
prefer function over method where you can

Reply:
Added this to #293. I'm planning to rewrite this function to work with the drives API and also allow for distant dependent connection probabilities

Reply:
nvm, just moved it to a function as the previous comment doesn't really impact that decision